src/net/ossl.cc

```diff
@@ -1084,17 +1118,21 @@ class session : public enable_shared_from_this<session>, public session_impl {
     // This function will attempt to pull data off of the _in stream
     // if there isn't already data needing to be processed first.
     future<> wait_for_input() {
+        tls_log.trace("{} wait_for_input", *this);
         // If we already have data, then it needs to be processed
         if (!_input.empty()) {
+            tls_log.trace("{} wait_for_input: input not empty", *this);
             return make_ready_future();
         }
         return _in.get()
           .then([this](buf_type buf) {
               // Set EOF if it's empty
+              tls_log.debug("{} wait_for_input: buffer {}empty", *this, buf.empty() ? "is ": "");
```

Comment:
i don't really understand this logging message:

- if `buf` is empty, we print "wait_for_input: buffer is empty" -- looks good
- if `buf` is not empty, we print "wait_for_input: buffer empty" -- what does this mean? shall we print "wait_for_input: buffer is not empty" or "wait_for_input: buffer not empty" instead?